File: iosApp/iosApp/BackendDocumentRepository.swift
import Foundation
import FirebaseFirestore

struct AppDocument: Identifiable {
    let id: String
    let name: String
    let htmlContent: String
    let createdAt: Date?
    let updatedAt: Date?
}

class BackendDocumentRepository {
    private let db = Firestore.firestore()

    func getAllDocuments() async throws -> [AppDocument] {
        let snapshot = try await db.collection("Documents").getDocuments()
        return snapshot.documents.map { doc in
            AppDocument(
                id: doc.documentID,
                name: doc.data()["name"] as? String ?? "",
                htmlContent: doc.data()["htmlContent"] as? String ?? "",
                createdAt: (doc.data()["createdAt"] as? Timestamp)?.dateValue(),
                updatedAt: (doc.data()["updatedAt"] as? Timestamp)?.dateValue()
            )
        }.sorted { $0.name < $1.name }
    }
}

File: composeApp/src/androidMain/kotlin/com/humblesolutions/indsphinx/repository/BackendDocumentRepository.kt
package com.humblesolutions.indsphinx.repository

import com.google.firebase.firestore.FirebaseFirestore
import com.humblesolutions.indsphinx.model.Document
import kotlinx.coroutines.tasks.await

class BackendDocumentRepository : DocumentRepository {
    private val db = FirebaseFirestore.getInstance()

    override suspend fun getAllDocuments(): List<Document> {
        val snapshot = db.collection("Documents").get().await()
        return snapshot.documents.map { doc ->
            Document(
                id = doc.id,
                name = doc.getString("name") ?: "",
                htmlContent = doc.getString("htmlContent") ?: "",
                createdAt = doc.getTimestamp("createdAt")?.toDate()?.time ?: 0L,
                updatedAt = doc.getTimestamp("updatedAt")?.toDate()?.time ?: 0L
            )
        }.sortedBy { it.name }
    }
}
